File: backend/models/blog.py
from typing import Optional

from pydantic import BaseModel


class BlogPostBase(BaseModel):
    title: str
    titlePt: str
    excerpt: str
    excerptPt: str
    content: str
    contentPt: str
    date: str
    tag: str
    author: str
    coverImage: Optional[str] = None


class BlogPostCreate(BlogPostBase):
    pass


class BlogPostUpdate(BlogPostBase):
    pass


class BlogPostOut(BlogPostBase):
    id: str

File: frontend/src/services/blog.ts
import { api } from "@/lib/api";

export interface BlogPost {
  id: string;
  title: string;
  titlePt: string;
  excerpt: string;
  excerptPt: string;
  content: string;
  contentPt: string;
  date: string;
  tag: string;
  author: string;
  coverImage?: string;
}

export type BlogPostInput = Omit<BlogPost, "id">;

export const blogService = {
  list: () => api.get<BlogPost[]>("/api/v1/blog"),
  get: (id: string) => api.get<BlogPost>(`/api/v1/blog/${id}`),
  create: (data: BlogPostInput) => api.post<BlogPost>("/api/v1/blog", data),
  update: (id: string, data: BlogPostInput) => api.put<BlogPost>(`/api/v1/blog/${id}`, data),
  remove: (id: string) => api.delete<void>(`/api/v1/blog/${id}`),
};
